manim/mobject/value_tracker.py

```diff
@@ -71,76 +75,102 @@ def construct(self):
 
     """
 
-    def __init__(self, value: float = 0, **kwargs: Any):
+    def __init__(self, value: float = 0, **kwargs: Any) -> None:
         super().__init__(**kwargs)
         self.set(points=np.zeros((1, 3)))
         self.set_value(value)
 
     def get_value(self) -> float:
         """Get the current value of this ValueTracker."""
-        return self.points[0, 0]
+        value: float = self.points[0, 0]
+        return value
 
-    def set_value(self, value: float):
-        """Sets a new scalar value to the ValueTracker"""
+    def set_value(self, value: float) -> Self:
+        """Sets a new scalar value to the ValueTracker."""
         self.points[0, 0] = value
         return self
 
-    def increment_value(self, d_value: float):
-        """Increments (adds) a scalar value  to the ValueTracker"""
+    def increment_value(self, d_value: float) -> Self:
+        """Increments (adds) a scalar value to the ValueTracker."""
         self.set_value(self.get_value() + d_value)
         return self
 
-    def __bool__(self):
-        """Return whether the value of this value tracker evaluates as true."""
+    def __bool__(self) -> bool:
+        """Return whether the value of this ValueTracker evaluates as true."""
         return bool(self.get_value())
 
-    def __iadd__(self, d_value: float):
-        """adds ``+=`` syntax to increment the value of the ValueTracker"""
-        self.increment_value(d_value)
+    def __iadd__(self, d_value: float | Mobject) -> Self:
+        """adds ``+=`` syntax to increment the value of the ValueTracker."""
+        if isinstance(d_value, Mobject):
+            raise ValueError(
+                "Cannot increment ValueTracker by a Mobject. Please provide a scalar value."
+            )
+        else:
+            self.increment_value(d_value)
```

Comment:
Minor nitpick: the `else` branch is not necessary, since the main `if` branch ends with an error. Having the `return self` statement outside of the `else` branch looks slightly weird in this case, so I'd suggest deleting the `else`.
```suggestion
        self.increment_value(d_value)
```